src/main/fc/rc.c

```diff
@@ -572,55 +567,66 @@ FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, int axis)
     prevSetpointSpeed[axis] = setpointSpeed;
 
-    // smooth the acceleration element (effectively a second order filter) and apply jitter reduction
+    // smooth the acceleration element (effectively a second order filter since incoming setpoint was smoothed)
+    // apply jitter reduction to acceleration here, effectively twice on acceleration
     setpointAcceleration = prevAcceleration[axis] + pid->feedforwardSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
     prevAcceleration[axis] = setpointAcceleration * zeroTheAcceleration;
     setpointAcceleration = setpointAcceleration * pid->feedforwardBoostFactor * jitterAttenuator * zeroTheAcceleration;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 1, lrintf(setpointSpeed * 0.1f)); // base feedforward without acceleration
-    }
+    feedforward = setpointSpeed + setpointAcceleration;
 
-    float feedforward = setpointSpeed + setpointAcceleration;
+    // apply jitter attenuation to classic feedforward elements only (twice on acceleaertion)
+    feedforward *= jitterAttenuator;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 2, lrintf(feedforward * 0.1f));
-        // un-smoothed feedforward including acceleration but before limiting, transition, averaging, and jitter reduction
-    }
-
-    // apply feedforward transition
+    // apply feedforward transition to classic feedforward elements only
     const bool useTransition = (pid->feedforwardTransition != 0.0f) && (rcDeflectionAbs[axis] < pid->feedforwardTransition);
     if (useTransition) {
         feedforward *= rcDeflectionAbs[axis] * pid->feedforwardTransitionInv;
     }
 
-    // apply averaging
-    if (feedforwardAveraging) {
-        feedforward = laggedMovingAverageUpdate(&feedforwardDeltaAvg[axis].filter, feedforward);
+    // apply max rate limiting, forcing to zero when sticks approach max, but not on yaw
+    if (axis < FD_YAW) {
+        if (pid->feedforwardMaxRateLimit) {
+            if (feedforward * setpoint > 0.0f) { // in same direction
+                const float limit = (maxRcRate[axis] - fabsf(setpoint)) * pid->feedforwardMaxRateLimit;
+                feedforward = (limit > 0.0f) ? constrainf(feedforward, -limit, limit) : 0.0f;
+            }
+        }
+    } else {
+        // add high-pass filtered setpoint to feedforward for yaw
+        // this is not a derivative, being proportional to setpoint, so does not require jitter reduction
+        // it is not interpolated, so dropouts will cause flat spots, but no derivative drops to zero
+        // this provides a sustained FF on yaw that mimics the normal yaw motor drive requirements
+        static float prevSetpointYaw = 0.0f;
+        // const of 7.0 to get decay roughly right
+        const float setpointLpfYaw = prevSetpointYaw + 7.0f * rxInterval * (setpoint - prevSetpointYaw);
+        prevSetpointYaw = setpointLpfYaw;
+        // const of 20 to scale yaw setpoint factor to a decent number
```

Comment:
And more magic :-(

Reply:
This is a 'for testing' PR :-)

Once we get logs that show whether the current estimate of a suitable value  (eg 7.0) is optimal, we can then consider if this smoothing factor can be a defined constant, or externalised to CLI, if user adjustment is required.

Some 'magic numbers' are needed to set the 'ballpark' value at this point.